db/database.go

```diff
@@ -293,6 +295,14 @@ func ConnectToBucket(spec base.BucketSpec) (base.Bucket, error) {
 	return ibucket.(base.Bucket), nil
 }
 
+// GetConnectToBucketFn returns a different OpenBucketFn to connect to the bucket depending on the value of failFast
+func GetConnectToBucketFn(failFast bool) OpenBucketFn {
+	if failFast {
+		return ConnectToBucketFailFast
+	}
+	return ConnectToBucket
+}
```

Comment:
I think we should make these `ConnectToBucket(FailFast)?` functions private so their use is restricted - encourage people to use this `GetConnectToBucketFn`